src/m365/spfx/commands/project/project-azuredevops-pipeline-add.ts

```diff
@@ -0,0 +1,213 @@
+import GlobalOptions from '../../../../GlobalOptions.js';
+
+class SpfxProjectAzureDevOpsPipelineAddCommand extends BaseProjectCommand {
+  private static loginMethod: string[] = ['application', 'user'];
+  private static scope: string[] = ['tenant', 'sitecollection'];
+  public static ERROR_NO_PROJECT_ROOT_FOLDER: number = 1;
+
+  public get name(): string {
+    return commands.PROJECT_AZUREDEVOPS_PIPELINE_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a Azure DevOps Pipeline for a SharePoint Framework project.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        name: typeof args.options.name !== 'undefined',
+        branchName: typeof args.options.branchName !== 'undefined',
+        loginMethod: typeof args.options.loginMethod !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined',
+        skipFeatureDeployment: !!args.options.skipFeatureDeployment
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-n, --name [name]'
+      },
+      {
+        option: '-b, --branchName [branchName]'
+      },
+      {
+        option: '-l, --loginMethod [loginMethod]',
+        autocomplete: SpfxProjectAzureDevOpsPipelineAddCommand.loginMethod
+      },
+      {
+        option: '-s, --scope [scope]',
+        autocomplete: SpfxProjectAzureDevOpsPipelineAddCommand.scope
+      },
+      {
+        option: '-u, --siteUrl [siteUrl]'
+      },
+      {
+        option: '--skipFeatureDeployment'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.scope && args.options.scope === 'sitecollection') {
+          if (!args.options.siteUrl) {
+            return `siteUrl option has to be defined when scope set to ${args.options.scope}`;
+          }
+
+          const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.siteUrl);
+          if (isValidSharePointUrl !== true) {
+            return isValidSharePointUrl;
+          }
+        }
+
+        if (args.options.loginMethod && SpfxProjectAzureDevOpsPipelineAddCommand.loginMethod.indexOf(args.options.loginMethod) < 0) {
+          return `${args.options.loginMethod} is not a valid login method. Allowed values are ${SpfxProjectAzureDevOpsPipelineAddCommand.loginMethod.join(', ')}`;
+        }
+
+        if (args.options.scope && SpfxProjectAzureDevOpsPipelineAddCommand.scope.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpfxProjectAzureDevOpsPipelineAddCommand.scope.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
I don't think is necessary. Seems like we're using both args

Reply:
oh sorry, I think it was some left-over